caustic/lenses/tnfw.py

```diff
@@ -0,0 +1,303 @@
+    def physical_deflection_angle(
+            self, x: Tensor, y: Tensor, z_s: Tensor, z_l, x0, y0, m, c, t, *args, params: Optional["Packed"] = None, **kwargs
+    ) -> tuple[Tensor, Tensor]:
+        """Compute the physical deflection angle (arcsec) for this lens at
+        the requested position. Note that the NFW profile is more
+        naturally represented as a physical deflection angle, this is
+        easily internally converted to a reduced deflection angle.
+
+        Args:
+            z_l (Tensor): Redshift of the lens.
+            x0 (Tensor): Center of lens position on x-axis (arcsec). 
+            y0 (Tensor): Center of lens position on y-axis (arcsec). 
+            m (Tensor): Mass of the lens (Msol).
+            c (Tensor): Concentration parameter of the lens (r200/rs for a classic NFW).
+            t (Tensor): Truncation scale. Ratio of truncation radius to scale radius (rt/rs).
+            params (dict): Dynamic parameter container.
+
+        Returns:
+            tuple[Tensor, Tensor]: The physical deflection angles in the x and y directions (arcsec).
+
+        """
+        d_l = self.cosmology.angular_diameter_distance(z_l, params)
+        rs = self.get_scale_radius(params)
+        x, y = translate_rotate(x, y, x0, y0)
+        r = ((x**2 + y**2).sqrt() + self.s) * d_l * arcsec_to_rad
+        theta = torch.arctan2(y,x)
+
+        # The below actually equally comes from eq 2.13 in Meneghetti notes
+        dr = self.projected_mass(r, z_s, params) / r # note dpsi(u)/du = 2x*dpsi(x)/dx when u = x^2
+        S = 4 * G_over_c2 * rad_to_arcsec
+        return S * dr * theta.cos(), S * dr * theta.sin()
+
+    @unpack(3)
+    def potential(
+        self, x: Tensor, y: Tensor, z_s: Tensor, z_l, x0, y0, c, t, *args, params: Optional["Packed"] = None, **kwargs
+    ) -> Tensor:
+        """
+        Compute the lensing potential. Note that this is not a unitless potential! This is the potential as given in Baltz et al. 2008.
+
+        TODO: convert to dimensionless potential.
+
+        Args:
+            z_l (Tensor): Redshift of the lens.
+            x0 (Tensor): Center of lens position on x-axis (arcsec). 
+            y0 (Tensor): Center of lens position on y-axis (arcsec). 
+            m (Tensor): Mass of the lens (Msol).
+            c (Tensor): Concentration parameter of the lens (r200/rs for a classic NFW).
+            t (Tensor): Truncation scale. Ratio of truncation radius to scale radius (rt/rs).
+            params (dict): Dynamic parameter container.
+
+        Returns:
+            Tensor: The lensing potential.
+        """
+        x, y = translate_rotate(x, y, x0, y0)
+        r = (x**2 + y**2).sqrt() + self.s
+        rs = self.get_scale_radius(params)
+        x = r / rs
+        u = x**2
+        F = self._F(x)
+        L = self._L(x, t)
+
+        #d_l = self.cosmology.angular_diameter_distance(z_l, params)
+        S = 2 * self.get_M0(params) * G_over_c2# * rad_to_arcsec * d_l**2
```

Comment:
Maybe I am completely wrong, but isn't the dimensionless potential phi(x)=phi(r)/r_s? i.e. S only needs to be divided by the scale radius r_s in Mpc?

Reply:
I tried dividing by `r_s` and that didn't line it up with lenstronomy. That's not a perfect test, but certainly it means more work needed.